<launch>
  <arg name="monitor_robot_state" default="false"/>
  <arg name="allow_trajectory_execution" default="false"/>

  <param name="robot_description" textfile="$(find planning_models)/test/urdf/robot.xml" />
  <param name="robot_description_semantic" textfile="$(find planning_models)/test/srdf/robot.xml" />

  <rosparam if="$(arg allow_trajectory_execution)" command="load" file="$(find pr2_arm_navigation)/config/pr2_both_arms_controllers.yaml" />

  <group ns="robot_description_planning">
    <rosparam command="load" file="$(find pr2_moveit_config)/config/collision_matrix.yaml"/>
    <rosparam command="load" file="$(find pr2_arm_navigation)/config/joint_limits.yaml"/>
  </group>
  
  <node unless="$(arg monitor_robot_state)" pkg="robot_state_publisher" type="state_publisher" name="rob_st_pub" />
  <node unless="$(arg monitor_robot_state)" pkg="moveit_visualization_ros" name="wall_clock_server" type="fake_time.py" />

  <node pkg="assisted_teleop" type="assisted_teleop_node" name="assisted_teleop_node" output="screen">
    <rosparam command="load" file="$(find pr2_moveit_config)/config/ompl_planning.yaml"/>
    <rosparam command="load" file="$(find pr2_moveit_config)/config/kinematics.yaml"/>
    <rosparam if="$(arg allow_trajectory_execution)" file="$(find pr2_arm_navigation)/config/pr2_execution_controllers.yaml"/>
    <param name="monitor_robot_state" value="$(arg monitor_robot_state)"/>
    <param name="allow_trajectory_execution" value="$(arg allow_trajectory_execution)"/>
  </node>

</launch>
